networking/p2p/src/main/java/tech/pegasys/artemis/networking/p2p/MothraP2PNetwork.java

```diff
@@ -61,23 +64,34 @@ public MothraP2PNetwork(
       String networkInterface,
       String identity,
       String bootnodes,
-      boolean isBootnode) {
+      boolean isBootnode,
+      List<String> peers) {
     this.eventBus = eventBus;
     this.store = store;
     this.port = port;
     this.networkInterface = networkInterface;
     this.identity = identity;
     this.bootnodes = bootnodes;
     this.isBootnode = isBootnode;
+    this.peers = peers;
     this.gossipProtocol = GossipProtocol.GOSSIPSUB;
     eventBus.register(this);
     this.handler = new MothraHandler(this.eventBus, this.store);
     mothra.Init();
     mothra.DiscoveryMessage = this.handler::handleDiscoveryMessage;
     mothra.ReceivedGossipMessage = this.handler::handleGossipMessage;
     mothra.ReceivedRPCMessage = this.handler::handleRPCMessage;
+    if (this.peers.size() == 0) {
+      // TODO - issue #827:
+      //      Once i have a reliable way to be notified when
+      //      libp2p peers are found then this can be removed
+      try {
+        Thread.sleep(15000);
```

Comment:
Probably a stupid question, but this doesn't delay the entire event bus, right?

Reply:
it needs to go away.  i only use it with discovery.  i am just giving it enough time to detect some peers before the chain starts.  i have an event that bubbles up from mothra, but havent done anything with it yet.